### ISSUE TITLE
Splitting dist and documentation paths in uri

### ISSUE THREAD

**Issue by jbergstroem** [nodejs/build#2376]
We currently have multiple ways of linking to and hosting both downloads and documentation. 

Targeting documentation alone, one linked from front page and one from docs
- https://nodejs.org/dist/latest-v12.x/docs/api/ and
- https://nodejs.org/docs/latest-v12.x/api/

If you check [our nginx config](https://raw.githubusercontent.com/nodejs/build/master/ansible/www-standalone/resources/config/nodejs.org) you can see that `/dist/`, `/docs/` and `/download/` is more or less identical for these reasons.

I would like to re-introduce `dist.nodejs.org` (which was a thing in the past) and `docs.nodejs.org`. The reasons are purely technical (some may have preferences to how url's are designed); it makes it easier for us to do things like traffic shaping and routing. It can open the doors to having docs fully self-maintained and not even tied to our `www` server.

My main task would be starting with downloads where we would have to consider things like redirecting clients to the new location. I checked in with @ljharb of `nvm` fame which said that it is already covered with how `curl` and previously `wget` was invoked (read: following redirects). @rvagg mentioned that there _might_ be clients that doesn't follow redirects. Does anyone know about use-cases like these?

**Comment by rvagg** [comment]
It's going to be in enterprise workflows where they're using `curl` without `-L`, and we're not likely to see them until we break stuff unfortunately! We could do this slow-ish, by first enabling it in parallel to the other options, and then trying it over a weekend, and then trying it over a week where there's a new release, and then turning it on full-time.

**Comment by jbergstroem** [comment]
> We could do this slow-ish, by first enabling it in parallel to the other options, and then trying the redirect over a weekend, and then trying it over a week where there's a new release, and then turning it on full-time.

If we can catch 50% of the traffic, that's a great improvement.
